// @uiw/react-codemirror/cjs/theme/dimensionTheme.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getDimensionTheme = getDimensionTheme;
exports.scrollerTheme = void 0;
var _view = require("@codemirror/view");
var scrollerTheme = exports.scrollerTheme = _view.EditorView.theme({
  '& .cm-scroller': {
    height: '100% !important'
  }
});
var lastDimensionKey = null;
var lastDimensionTheme = null;
function getDimensionTheme(height, minHeight, maxHeight, width, minWidth, maxWidth) {
  if (!height && !minHeight && !maxHeight && !width && !minWidth && !maxWidth) {
    return null;
  }
  var cacheKey = JSON.stringify({
    height: height,
    minHeight: minHeight,
    maxHeight: maxHeight,
    width: width,
    minWidth: minWidth,
    maxWidth: maxWidth
  });
  if (cacheKey === lastDimensionKey) {
    return lastDimensionTheme;
  }
  lastDimensionKey = cacheKey;
  lastDimensionTheme = _view.EditorView.theme({
    '&': {
      height: height,
      minHeight: minHeight,
      maxHeight: maxHeight,
      width: width,
      minWidth: minWidth,
      maxWidth: maxWidth
    }
  });
  return lastDimensionTheme;
}
